gui/src/components/mainInput/TipTapEditor.tsx

```diff
@@ -596,38 +745,46 @@ const TipTapEditor = memo(function TipTapEditor({
       if (!json.content?.some((c) => c.content)) {
         return;
       }
-
+  
+      if (historyLength === 0 && contextItems[0]?.id.providerTitle === "file") {
+        const fileContextItem = contextItems[0];
+        const mentionNode = {
+          type: 'mention',
+          attrs: {
+            id: fileContextItem.description,
+            label: fileContextItem.name.replace(/^@/, ''),
+            renderInlineAs: null,
+            query: fileContextItem.description,
+            itemType: "file"
+          }
+        };
+  
+        json.content.push({
+          type: 'paragraph',
+          content: [mentionNode]
+        });
+      }
+  
       onEnter(json, modifiers);
-
+  
       if (isMainInput) {
         const content = editor.state.toJSON().doc;
         addRef.current(content);
         editor.commands.clearContent(true);
       }
     },
-    [onEnter, editor, isMainInput],
+    [onEnter, editor, isMainInput, historyLength, contextItems],
   );
 
-  // This is a mechanism for overriding the IDE keyboard shortcut when inside of the webview
-  const [ignoreHighlightedCode, setIgnoreHighlightedCode] = useState(false);
-
   useEffect(() => {
-    const handleKeyDown = (event: any) => {
-      if (
-        isMetaEquivalentKeyPressed(event) &&
-        (isJetBrains() ? event.code === "KeyJ" : event.code === "KeyL")
-      ) {
-        setIgnoreHighlightedCode(true);
```

Comment:
ignoreHighlightedCode and setIgnoreHighlightedCode were removed because they were no longer relevant and not needed.